motionplan/ik/nloptInverseKinematics.go

```diff
@@ -101,20 +139,16 @@ func (ik *NloptIK) Solve(ctx context.Context,
 	if len(ik.lowerBound) == 0 || len(ik.upperBound) == 0 {
 		return errBadBounds
 	}
-	mInput := &State{Frame: ik.model}
 	var activeSolvers sync.WaitGroup
 
-	// x is our joint positions
+	// x is our set of inputs
 	// Gradient is, under the hood, a unsafe C structure that we are meant to mutate in place.
 	nloptMinFunc := func(x, gradient []float64) float64 {
 		iterations++
 
-		// Requesting an out-of-bounds transform will result in a non-nil error but will optionally return a correct if invalid pose.
-		// Thus we check if eePos is nil, and if not, continue as normal and ignore errors.
-		// As confirmation, the "input out of bounds" string is checked for in the error text.
 		inputs := referenceframe.FloatsToInputs(x)
 		eePos, err := ik.model.Transform(inputs)
-		if eePos == nil || (err != nil && !strings.Contains(err.Error(), referenceframe.OOBErrString)) {
+		if eePos == nil || err != nil {
```

Comment:
Why is this check no longer necessary?

Reply:
Eek, I meant to remove the other one where Ray commented; restoring this.